lib/schedule/piece.ex

```diff
@@ -53,42 +51,27 @@ defmodule Schedule.Piece do
     end_mid_route?: false
   ]
 
-  @spec hydrate(t(), Trip.by_id(), Timepoint.timepoint_names_by_id()) :: t()
-  def hydrate(piece, trips_by_id, timepoint_names_by_id) do
-    trip_ids = piece.trips
-
-    trips =
-      Enum.map(trip_ids, fn
-        trip_id when is_binary(trip_id) ->
-          trips_by_id[trip_id]
-
-        %Trip{} = trip ->
-          trip
-
-        %AsDirected{} = as_directed ->
-          as_directed
-      end)
-      |> Enum.map(&Trip.set_pretty_names(&1, timepoint_names_by_id))
-
-    %{
-      piece
-      | trips: trips,
-        start_place: Timepoint.pretty_name_for_id(timepoint_names_by_id, piece.start_place),
-        end_place: Timepoint.pretty_name_for_id(timepoint_names_by_id, piece.end_place),
-        start_mid_route?:
-          piece.start_mid_route? &&
-            hydrate_mid_route_swing(piece.start_mid_route?, trips_by_id, timepoint_names_by_id)
-    }
+  @spec block_key(t()) :: Block.key()
+  def block_key(piece) do
+    {piece.schedule_id, piece.block_id}
   end
 
-  @spec hydrate_mid_route_swing(
-          mid_route_swing(),
-          Schedule.Trip.by_id(),
-          Timepoint.timepoint_names_by_id()
-        ) :: mid_route_swing()
-  defp hydrate_mid_route_swing(mid_route_swing, trips_by_id, timepoint_names_by_id) do
-    trip_id = mid_route_swing.trip
-    trip = trips_by_id |> Map.get(trip_id) |> Trip.set_pretty_names(timepoint_names_by_id)
-    %{mid_route_swing | trip: trip}
+  # When we prepare a HASTUS export for a new rating for Skate, the export's
+  # data starts not with the start date of the new rating, but rather the
+  # current date. This allows us to import the new rating data as soon as it's
+  # ready, rather than having to try to import it at the moment the new rating
+  # is starting. But it also means that, at any given moment, Skate's HASTUS
+  # data contains a significant number of trips that belong to either a past
+  # or future rating.
+  #
+  # These trips will have a nil service_id. So do all non-revenue trips, but
+  # since a piece should always have at least one revenue trip, if all trips
+  # in a piece have a nil service_id, we can assume that all those trips are
+  # from a non-current rating.
+  @spec from_non_current_rating?(t()) :: boolean()
+  def from_non_current_rating?(piece) do
+    piece.trips
+    |> Enum.reject(&match?(%AsDirected{}, &1))
+    |> Enum.all?(&is_nil(&1.service_id))
```

Comment:
Question: I know trips from a past rating won't have a `service_id` (at least once the old trips are gone from GTFS), but what about trips from a future rating (once it's in GTFS at least?). There are two possibilities here:
1. These trips will have a `service_id`, in which case `from_non_current_rating?` is maybe a _slightly_ weird name... `from_non_past_rating?` perhaps? Though as I pointed out trips from a future rating might not have a `service_id` _yet_ if the rating hasn't hit GTFS...
2. These trips won't have a `service_id`, in which case we could have a problem with how the application handles rolling over on the first day of a new rating, and we'd somehow have to trigger a re-load of the schedule data.

Reply:
excellent question, @dks-mbta would you know this one too?

Reply:
Well, I think this is more of a question about how Skate processes the data than it is about the HASTUS export or GTFS itself, right? We know that HASTUS will include the trips for the present and future rating, and we know that the trips for the future rating will (eventually) be in GTFS, so it's a question of whether or not they get paired up with a service ID like trips in the current rating do.

Reply:
OK I think I misunderstood your question at first. But suppose that today is during rating N and the data for rating N+1 just got added to GTFS, Skate rebooted and ingested the new data, and those trips don't have `service_id`s. Those trips will get filtered out for now: fine. Unless I'm missing something, though, at some point before rating N+1 actually starts, someone is going to have to update the GTFS so that these trips _do_ have `service_id`s, because all revenue trips for a current rating are supposed to have one.

So we really only have to consider case #1 there, and a name change makes sense then. Does that all add up, or do you see anything wrong with my reasoning?

Reply:
> Unless I'm missing something, though, at some point before rating N+1 actually starts, someone is going to have to update the GTFS so that these trips do have service_ids, because all revenue trips for a current rating are supposed to have one.

I actually think the trips will have a service ID in the GTFS from the start (it's a required field per [the spec](http://gtfs.org/reference/static#tripstxt)). In which case... I think we're fine? Again, the main way it could go wrong is on the Skate end if we somehow filter some of that information out, but I don't believe we currently do.

Reply:
I'll change the name anyhow, that makes sense to me.